nexus/src/app/instance_platform/cpu_platform.rs

```diff
@@ -0,0 +1,870 @@
+    leaf.set_dtlb_2m_4m_size(0x800);
+    leaf.set_dtlb_2m_4m_associativity(0x4);
+
+    // Set up leaf 8000_0006h EBX
+    leaf.set_itlb_4k_size(0x200);
+    leaf.set_itlb_4k_associativity(0x4);
+    leaf.set_dtlb_4k_size(0x800);
+    leaf.set_dtlb_4k_associativity(0x6);
+
+    // Set up leaf 8000_0006h ECX
+    leaf.set_l2cache_line_size(0x40);
+    leaf.set_l2cache_lines_per_tag(0x1);
+    leaf.set_l2cache_associativity(0x6);
+    leaf.set_l2cache_size(0x0200);
+
+    // Set up leaf 8000_0006h EDX
+    leaf.set_l3cache_line_size(0x40);
+    leaf.set_l3cache_lines_per_tag(0x1);
+    leaf.set_l3cache_associativity(0x9);
+    leaf.set_l3cache_size(0x0200);
+
+    cpuid
+        .set_l2_l3_cache_and_tlb_info(Some(leaf))
+        .expect("can set leaf 8000_0006h");
+
+    // Set up TLB information for 1GiB pages (leaf 8000_0019h)
+    let mut leaf = Tlb1gbPageInfo::empty();
+    leaf.set_dtlb_l1_1gb_associativity(0xF);
+    leaf.set_dtlb_l1_1gb_size(0x40);
+    leaf.set_itlb_l1_1gb_associativity(0xF);
+    leaf.set_itlb_l1_1gb_size(0x40);
+    leaf.set_dtlb_l2_1gb_associativity(0xF);
+    leaf.set_dtlb_l2_1gb_size(0x40);
+    leaf.set_itlb_l2_1gb_associativity(0);
+    leaf.set_itlb_l2_1gb_size(0);
+    cpuid.set_tlb_1gb_page_info(Some(leaf)).expect("can set leaf 8000_0019h");
+
+    // Set up extended cache hierarchy info (leaf 8000_001Dh)
+    let mut levels = Vec::new();
+    levels.push(CpuIdResult {
+        eax: 0x00000121,
+        ebx: 0x01C0003F,
+        ecx: 0x0000003F,
+        edx: 0x00000000,
+    });
+    levels.push(CpuIdResult {
+        eax: 0x00000122,
+        ebx: 0x01C0003F,
+        ecx: 0x0000003F,
+        edx: 0x00000000,
+    });
+    levels.push(CpuIdResult {
+        eax: 0x00000143,
+        ebx: 0x01C0003F,
+        ecx: 0x000003FF,
+        edx: 0x00000002,
+    });
+    levels.push(CpuIdResult {
+        eax: 0x00000163,
+        ebx: 0x03C0003F,
+        ecx: 0x00007FFF,
+        edx: 0x00000001,
+    });
+    cpuid
+        .set_extended_cache_parameters(Some(levels.as_slice()))
+        .expect("can set leaf 8000_001Dh");
+
+    dump_to_cpuid_entries(cpuid.into_source())
+}
+
+fn dump_to_cpuid_entries(dump: CpuIdDump) -> Vec<CpuidEntry> {
+    let mut entries = Vec::new();
+
+    for (leaf, subleaf, regs) in dump.into_iter() {
+        entries.push(CpuidEntry {
+            leaf: leaf,
+            subleaf: subleaf,
+            eax: regs.eax,
+            ebx: regs.ebx,
+            ecx: regs.ecx,
+            edx: regs.edx,
+        });
+    }
+
+    // Entry order does not actually matter. Sort here because it's fast (~30-35 leaves) and
+    // looking at the vec in logs or on the wire *so* much nicer.
+    entries.sort_by(|left, right| {
+        let by_leaf = left.leaf.cmp(&right.leaf);
+        if by_leaf == std::cmp::Ordering::Equal {
+            left.subleaf.cmp(&right.subleaf)
+        } else {
+            by_leaf
+        }
+    });
+
+    entries
+}
+
+#[test]
+fn milan_rfd314_is_as_described() {
```

Comment:
the other cool thing we can do now is grab a CPUID dump from a guest before using explicit definitions and include a test describing how it and the RFD 314 definitions differ. i want to include that alongside this test before merging (would have been _so_ much nicer than the manual comparison i did before...)